### PR TITLE
adopt PEP621

### PR DESCRIPTION
**Pull request by MoLow**
Fixes: https://github.com/nodejs/tap2junit/issues/39

### CODE REVIEW

**Comment by MoLow** [comment]
CC @cclauss

File: .github/workflows/test.yml
Comment (cclauss) [inline .github/workflows/test.yml:1]: Please undo this name change.

Comment (MoLow) [reply .github/workflows/test.yml:1]: why? this workflow does more than just linting

File: .github/workflows/test.yml
Comment (cclauss) [inline .github/workflows/test.yml:11]: If we are going to go this way then we should make a full https://pre-commit.com so that tests are run the same locally and in CI

File: pyproject.toml
Comment (cclauss) [inline pyproject.toml:15]: NOT!!!

File: pyproject.toml
Comment (cclauss) [inline pyproject.toml:18]: NOT!!!

File: pyproject.toml
Comment (cclauss) [inline pyproject.toml:20]: Please add 3.9 and 3.10

File: pyproject.toml
Comment (cclauss) [inline pyproject.toml:48]: Please add a blank line at the end of the file.

File: pyproject.toml
Comment (cclauss) [inline pyproject.toml:34]:
```suggestion
```

File: .github/workflows/test.yml
Comment (MoLow) [reply .github/workflows/test.yml:11]: why is this related? correct me if I am wrong, but prior to this PR there was no pre-commit hook

Comment (cclauss) [reply .github/workflows/test.yml:11]: There was none but if we are going to make `pip install tap2junit[test]` then we should provide a way to use all those testing dependencies locally.  If this is not done then there is noo reason to move the configuration out of the file that is running the tests.

Comment (MoLow) [reply .github/workflows/test.yml:11]: I see, I really have no strong opinion regarding this - do you prefer adding a pre-commit hook or moving these back into this file?

File: .github/workflows/test.yml
Comment (cclauss) [reply .github/workflows/test.yml:1]: It is more linting (bandit, codespell, flake8, mypy, safety) then testing (pytest) or reformatting (black, isort, pyupgrade).  `lint_python` is an Action that I broadly adopt to many projects so it is better to keep the naming unless we make substantial changes in the tools used.

File: .github/workflows/test.yml
Comment (cclauss) [reply .github/workflows/test.yml]: Let's do this...  We can land this as is and then I will circle back and create a pre-commit when we add support for Python 3.11 (due next month).

Comment (MoLow) [reply .github/workflows/test.yml]: 👍🏻

**Comment by MoLow** [comment]
Seems good to go, can we release a version after this lands?

**Review by cclauss — APPROVED** [review]
Nice work!  This helps a lot.

Basic testing done via:
`pipx install --force git+https://github.com/MoLow/tap2junit.git@migrate-to-use-pyproject_toml`

**Pull request merged**
